gtsam/base/concepts.h

```diff
@@ -13,7 +13,10 @@
 // to cause compilation errors.
 #ifdef COMPILE_ERROR_NOT_IMPLEMENTED
 
+#ifdef GTSAM_USE_BOOST_FEATURES
 #include <boost/static_assert.hpp>
+#endif
+
 #define CONCEPT_NOT_IMPLEMENTED BOOST_STATIC_ASSERT_MSG(boost::false_type, \
```

Comment:
If we conditionally exclude `<boost/static_assert.hpp>`, the macro defined in line 20 will become unavailable.

Reply:
In the meantime, I completely refactored concept checking to work with and without boost :-)